### PR TITLE
[REVIEW] Resolve unnecessary import of thrust/optional.hpp in types.hpp

### LINKED ISSUE #6475: [FEA] Remove `#include <thrust/optional.h>` from types.hpp
Currently, libcudf build fails if you remove the unused:
```cpp
#include <thrust/optional.h>  // TODO no idea why this is needed ¯\_(ツ)_/¯
```
from the file. The build needs to be fixed so that this doesn't happen.

### PR DESCRIPTION
**Pull request by vyasr**
Resolves #6475. The missing required include of `optional.hpp` was in `convert_datetime.cu`.

### CODE REVIEW

**Review by codereport — APPROVED** [review]
:fire: :fire: :fire: :fire:

**Comment by vyasr** [comment]
My local repo wasn't up-to-date, looks like #7461 introduced additional usage of optional relying on the include in types.hpp. Fixing now.

**Review by jrhemstad — APPROVED** [review]

**Comment by vyasr** [comment]
rerun tests

File: cpp/include/cudf/types.hpp [hunk 1]
```diff
@@ -24,8 +24,6 @@
 #define CUDA_DEVICE_CALLABLE inline
 #endif
 
-#include <thrust/optional.h>  // TODO no idea why this is needed ¯\_(ツ)_/¯
```
Comment (harrism) [inline cpp/include/cudf/types.hpp:27]: Need to update the copyright in this file as well to include 2021.

**Review by harrism — APPROVED** [review]
Nice find! Just one copyright needs updated (types.hpp)

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**